pkg/test/fake/client.go

```diff
@@ -24,51 +24,58 @@ func (d *CachedDiscovery) Invalidate() {
 
 }
 
-// CachedDiscoveryClient returns a fake discovery client that is populated with some types for use in
-// unit tests.
-func CachedDiscoveryClient() discovery.CachedDiscoveryInterface {
+func CustomCachedDiscoveryClient(additionalResources ...*metav1.APIResourceList) discovery.CachedDiscoveryInterface {
```

Comment:
I'd prefer to have `Fake` in the name e.g. `FakeCachedDiscoveryClient`.

Reply:
Well, the `fake` is in the package name, when you create it it's `fake.CachedDiscoveryClient` and `fake.CustomCachedDiscoveryClient`. If we were to add the `Fake` to the name it would stutter...

Reply:
You're right! Didn't see the `fake.` in the package path. 👍